provider/datagen/src/bin/datagen.rs

```diff
@@ -2,228 +2,212 @@
+    #[arg(
+        help = "Path to a local cldr-{version}-json-full.zip directory (see https://github.com/unicode-org/cldr-json/releases).\n\
+                  Note that some keys do not support versions before 41.0.0."
+    )]
+    cldr_root: Option<PathBuf>,
+
+    #[arg(long, value_name = "TAG", default_value = "latest")]
+    #[arg(
+        help = "Download Unicode Properties data from this GitHub tag (https://github.com/unicode-org/icu/tags)\n\
+                  Use 'latest' for the latest version verified to work with this version of the binary.\n\
+                  Ignored if '--icuexport-root' is present.\n\
+                  Note that some keys do not support versions before release-71-1."
+    )]
+    icuexport_tag: String,
+
+    #[arg(long, value_name = "PATH")]
+    #[arg(
+        help = "Path to a local icuexportdata_uprops_full directory (see https://github.com/unicode-org/icu/releases).\n\
+                  Note that some keys do not support versions before release-71-1."
+    )]
+    icuexport_root: Option<PathBuf>,
+
+    #[arg(long, value_enum, default_value_t = cli::TrieType::Small)]
+    #[arg(
+        help = "Whether to optimize CodePointTrie data structures for size (\"small\") or speed (\"fast\").\n\
+                  Using \"fast\" mode increases performance of CJK text processing and segmentation. For more\n\
+                  information, see the TrieType enum."
+    )]
+    trie_type: cli::TrieType,
+
+    #[arg(long, value_enum, default_value_t = cli::CollationHanDatabase::Implicit)]
+    #[arg(help = "Which collation han database to use.")]
+    collation_han_database: cli::CollationHanDatabase,
+
+    #[arg(long, value_enum, num_args = 1..)]
+    #[arg(
+        help = "Which less-common collation tables to include. 'search-all' includes all search tables."
+    )]
+    include_collations: Vec<cli::CollationTable>,
+
+    #[arg(long, hide = true)]
+    #[arg(help = "Deprecated, use --locales full or --locales modern")]
+    cldr_locale_subset: bool,
+
+    #[arg(long, short, num_args = 1..)]
+    #[arg(
+        help = "Include these resource keys in the output. Accepts multiple arguments.\n\
+                  Set to 'all' for all keys, 'experimental-all' to include experimental keys,\n\
+                  or 'none' for no keys."
+    )]
+    keys: Vec<String>,
+
+    #[arg(long, value_name = "KEY_FILE")]
+    #[arg(
+        help = "Path to text file with resource keys to include, one per line. Empty lines \
+                  and lines starting with '#' are ignored."
+    )]
+    key_file: Option<PathBuf>,
+
+    #[arg(long, value_name = "BINARY")]
+    #[arg(help = "Analyzes the binary and only includes keys that are used by the binary.")]
+    keys_for_bin: Option<PathBuf>,
+
+    #[arg(long, hide = true)]
+    #[arg(help = "Deprecated: alias for --keys all")]
+    all_keys: bool,
+
+    #[arg(long, short, required_unless_present = "all_locales", num_args = 0..)]
```

Comment:
This `required_unless_present` doesn't seem to make `locales` requires for `--help`.

Reply:
No, but that's always been the case IIRC.

Reply:
Currently the help text is 
```
icu4x-datagen 1.1.2
The ICU4X Project Developers
Learn more at: https://docs.rs/icu_datagen/1.1.2

USAGE:
    icu4x-datagen [FLAGS] [OPTIONS] --locales <LOCALES>... <--keys <KEYS>...|--key-file <KEY_FILE>|--keys-for-bin <KEYS_FOR_BIN>|--all-keys>

FLAGS:
        --fingerprint             --format=dir only: whether to add a fingerprints file to the output.
        --insert-feature-gates    --format=mod only: insert feature gates for individual `icu_*` crates. Requires --use-
                                  separate-crates
...
```

Reply:
The code is being changed from

```
        .arg(
            Arg::with_name("LOCALES")
                .short("l")
                .long("locales")
                .multiple(true)
                .takes_value(true)
                .required_unless("ALL_LOCALES")
                .help(
                    "Include this locale in the output. Accepts multiple arguments. \
                    Set to 'full' or 'modern' for the respective CLDR locale sets, or 'none' for no locales.",
                ),
        )
```

to

```
    #[arg(long, short, required_unless_present = "all_locales", num_args = 0..)]
    #[arg(
        help = "Include this locale in the output. Accepts multiple arguments. \
                  Set to 'full' or 'modern' for the respective CLDR locale sets, or 'none' for no locales."
    )]
    locales: Vec<String>,
```

which looks basically the same. So it seems like this would be a bug in clap if we want the arg to show as required?

Reply:
It's allowed to behave differently across major semver version, so I don't think it's a bug. We'll have to make changes if we want the help string to be correct (and we should, currently it looks like only `--keys` is required but then you get an error).